source/common/quic/envoy_quic_session_cache.cc

```diff
@@ -0,0 +1,168 @@
+#include "common/quic/envoy_quic_session_cache.h"
+
+namespace Envoy {
+namespace Quic {
+namespace {
+
+constexpr size_t kMaxSessionCacheEntries = 1024;
+
+// Returns false if the SSL |session| doesn't exist or it is expired at |system_time|.
+bool isSessionValid(SSL_SESSION* session, SystemTime system_time) {
```

Comment:
There is no `system_time_` member. I think you might be confusing it with `time_source_` which is different - we want to avoid computing the time from the time source too often.

Reply:
Makes sense. I've renamed `system_time` to `now`.